### PR TITLE
Move deprecation under use statements

### PR DESCRIPTION
**Pull request by greg0ire**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | none
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

**Review by aivus — APPROVED** [review]

**Comment by greg0ire** [comment]
Don't merge yet there are many more occurences.

**Comment by greg0ire** [comment]
Files I need to check:

- [ ] `src/Symfony/Component/Config/DependencyInjection/ConfigCachePass.php`
- [ ] `src/Symfony/Component/Console/Event/ConsoleExceptionEvent.php`
- [ ] `src/Symfony/Component/Process/ProcessBuilder.php`
- [ ] `src/Symfony/Component/Security/Core/Exception/NonceExpiredException.php`
- [ ] `src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MemcacheSessionHandler.php`
- [ ] `src/Symfony/Component/HttpFoundation/Session/Storage/Handler/WriteCheckSessionHandler.php`
- [ ] `src/Symfony/Bridge/Doctrine/HttpFoundation/DbalSessionHandler.php`
- [ ] `src/Symfony/Bridge/Doctrine/HttpFoundation/DbalSessionHandlerSchema.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CompilerDebugDumpPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TranslationExtractorPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/PropertyInfoPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TranslatorPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddConsoleCommandPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TranslationDumperPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddCacheClearerPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddCacheWarmerPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/SerializerPass.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/EventListener/TestSessionListener.php`
- [ ] `src/Symfony/Bundle/FrameworkBundle/Translation/PhpStringTokenParser.php`
- [ ] `src/Symfony/Bundle/TwigBundle/ContainerAwareRuntimeLoader.php`
- [ ] `src/Symfony/Bundle/TwigBundle/Command/DebugCommand.php`
- [ ] `src/Symfony/Bundle/SecurityBundle/Command/SetAclCommand.php`
- [ ] `src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php`
- [ ] `src/Symfony/Bundle/SecurityBundle/EventListener/AclSchemaListener.php`

found with `grep -rn trigger_error .|grep ':14:'|grep ':class'`

**Comment by greg0ire** [comment]
It's ready now

File: src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 namespace Symfony\Bundle\SecurityBundle\Command;
 
-@trigger_error(sprintf('Class "%s" is deprecated since version 3.4 and will be removed in 4.0. Use Symfony\Bundle\AclBundle\Command\SetAclCommand instead.', SetAclCommand::class), E_USER_DEPRECATED);
+@trigger_error(sprintf('Class "%s" is deprecated since version 3.4 and will be removed in 4.0. Use Symfony\Bundle\AclBundle\Command\SetAclCommand instead.', InitAclCommand::class), E_USER_DEPRECATED);
```
Comment (derrabus) [inline src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: I'd move this below the `use` statements and use `::class` for both classes.

**Comment by greg0ire** [comment]
Travis build failures seem unrelated.

File: src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php (hunk 1 above)
Comment (greg0ire) [reply src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: Wait my fix was incomplete it seems.

**Review by Simperfit — APPROVED** [review]

**Review by weaverryan — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Actually I think the convention has always been reverse: the deprecation should be just below the namespace. That's what we did in 2.7 & 2.8, and if you grep with a more generic regexp, you'll see that most of the existing notices are at the top.

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
should really be at the very top of the file, just below the namespace

**Comment by greg0ire** [comment]
@nicolas-grekas but then the use statements can't be used, see https://3v4l.org/8PjTg . I see 2 solutions:

1. reconsider the convention
2. refrain from using the `::class` notation when referring to other classes (is that what you meant?)

**Comment by nicolas-grekas** [comment]
Oh, got it! So this is a real bug fix. Let's merge as is then, no need to change everything now (and maybe discuss the general change somewhere else.)

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @greg0ire.

File: src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php [hunk 2]
```diff
@@ -22,6 +20,8 @@
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Schema\SchemaException;
 
+@trigger_error(sprintf('Class "%s" is deprecated since version 3.4 and will be removed in 4.0. Use Symfony\Bundle\AclBundle\Command\InitAclCommand instead.', InitAclCommand::class), E_USER_DEPRECATED);
+
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: Change is not needed

**Review by chalasr — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php (hunk 2 above)
Comment (chalasr) [reply src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: Race condition :) not a big deal

Comment (greg0ire) [reply src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: right, let me fix it

Comment (chalasr) [reply src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: Don't worry, it's merged already

Comment (greg0ire) [reply src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: done

**Pull request closed without merging**